app/scripts/directives/processTemplate.js

```diff
@@ -0,0 +1,364 @@
+  };
+
+  var showWarningsOrCreate = function(result) {
+    var alerts = SecurityCheckService.getSecurityAlerts(processedResources, ctrl.project.metadata.name);
+
+    // Now that all checks are completed, show any Alerts if we need to
+    var quotaAlerts = result.quotaAlerts || [];
+    alerts = alerts.concat(quotaAlerts);
+    var errorAlerts = _.filter(alerts, {type: 'error'});
+    if (errorAlerts.length) {
+      ctrl.disableInputs = false;
+      ctrl.precheckAlerts = alerts;
+    }
+    else if (alerts.length) {
+       launchConfirmationDialog(alerts);
+       ctrl.disableInputs = false;
+    }
+    else {
+      createResources();
+    }
+  };
+
+  ctrl.createFromTemplate = function() {
+    ctrl.disableInputs = true;
+    var userLabels = keyValueEditorUtils.mapEntries(keyValueEditorUtils.compactEntries(ctrl.labels));
+    var systemLabels = keyValueEditorUtils.mapEntries(keyValueEditorUtils.compactEntries(ctrl.systemLabels));
+    ctrl.template.labels = _.extend(systemLabels, userLabels);
+
+    DataService.create("processedtemplates", null, ctrl.template, context).then(
+      function(config) { // success
+        // Cache template parameters and message so they can be displayed in the nexSteps page
+        ProcessedTemplateService.setTemplateData(config.parameters, ctrl.template.parameters, config.message);
+        processedResources = config.objects;
+
+        QuotaService.getLatestQuotaAlerts(processedResources, context).then(showWarningsOrCreate);
+      },
+      function(result) { // failure
+        ctrl.disableInputs = false;
+        var details;
+        if (result.data && result.data.message) {
+          details = result.data.message;
+        }
+        ctrl.alerts["process"] =
+          {
+            type: "error",
+            message: "An error occurred processing the template.",
+            details: details
+          };
+      }
+    );
+  };
+
+  var shouldAddAppLabel = function() {
+    // If the template defines its own app label, we don't need to add one at all
+    if (_.get(ctrl.template, 'labels.app')) {
+      return false;
+    }
+
+    // Otherwise check if an object in the template has an app label defined
+    return !_.some(ctrl.template.objects, "metadata.labels.app");
+  };
+
+  function setTemplateParams() {
+    ctrl.parameterDisplayNames = {};
+    _.each(ctrl.template.parameters, function(parameter) {
+      ctrl.parameterDisplayNames[parameter.name] = parameter.displayName || parameter.name;
+    });
+
+    if(ctrl.prefillParameters) {
+      _.each(ctrl.template.parameters, function(parameter) {
+        if (ctrl.prefillParameters[parameter.name]) {
+          parameter.value = ctrl.prefillParameters[parameter.name];
+        }
+      });
+    }
+
+    findTemplateImages(ctrl.template);
+    var imageUsesParameters = function(image) {
+      return !_.isEmpty(image.usesParameters);
+    };
+    if (_.some(images, imageUsesParameters)) {
+      $scope.$watch(function() {
```

Comment:
did you double check this with the jenkins template to make sure its working? since this is the only bit still referencing $scope